ansible/roles/sunbird-system-init/templates/system_init_commands.j2

```diff
@@ -0,0 +1,336 @@
+        fieldName="$1";
+        fieldValue="$2";
+        if [ -z "$fieldValue" ];
+        then
+                echo "Mandatory field - ""$fieldName"" - is not configured. Exiting System Initialisation Program...";
+                exit 1;
+        fi
+}
+
+initializeVariables() {
+        sunbird_custodian_tenant_slug="$(echo -n ""$sunbird_custodian_tenant_channel"" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//' | sed -e 's/\s/-/g' | sed -e 's/[^a-zA-Z0-9-]//g' | tr -s '-' | tr A-Z a-z)"
+        currenttime=$(date +'%F %T:%3N%z')
+        if [ -z "$sunbird_custodian_tenant_slug" ] || [ -z "$currenttime" ];
+        then
+                echo "System Initialisation failed. Unable to initialise variables.";
+                exit 1;
+        fi
+}
+
+getExistingCustodianTenantId() {
+        sunbird_custodian_tenant_id=$(cqlsh -e "select id from sunbird.organisation where isrootorg=true and channel='""$sunbird_custodian_tenant_channel""' allow filtering;" | awk 'FNR == 4 {print}' | sed 's/ //g')
+}
+
+getNewCustodianTenantId() {
+        sunbird_custodian_tenant_id='0'$((($(date +%s%3N)+$(jot -r 1 0 999999))<<13))'0';
+}
+
+getValueFromSystemSettings() {
+        system_settings_id="$1";
+        system_settings_value=$(cqlsh -e "select value from sunbird.system_settings where id='""$system_settings_id""';" | awk 'FNR == 4 {print}' | sed 's/ //g')
+}
+
+setValueInSystemSettings() {
+        system_settings_id="$1";
+        system_settings_value="$2";
+        cqlsh -e "update sunbird.system_settings set value='""$system_settings_value""',field='""$system_settings_id""' where id='""$system_settings_id""';"
+}
+
+setPropertyInSystemSettings() {
+        system_settings_id="$1";
+        system_settings_field="$2";
+        system_settings_value="$3";
+        cqlsh -e "insert into sunbird.system_settings(id,field,value) values('""$system_settings_id""','""$system_settings_field""','""$system_settings_value""');"
+}
+
+registerChannel() {
+        register_channel_response_code=$(curl -o /dev/null -s -w "%{http_code}\n" \
+                -X POST $ekstep_proxy_base_url/api/channel/v3/create \
+                -H 'authorization: Bearer '"$ekstep_api_key" \
+                -H 'cache-control: no-cache' \
+                -H 'content-type: application/json' \
+                -d '{
+                        "request": {
+                                "channel": {
+                                        "name":"'"$sunbird_custodian_tenant_channel"'",
+                                        "description":"'"$sunbird_custodian_tenant_description"'",
+                                        "code":"'"$sunbird_custodian_tenant_id"'"
+                                }
+                        }
+                    }')
+}
+
+registerHashTag() {
+        register_hashtag_response_code=$(curl -o /dev/null -s -w "%{http_code}\n" \
+                -X POST $ekstep_proxy_base_url/api/data/v3/tag/register/$sunbird_custodian_tenant_id \
```

Comment:
Define variables for each URL as they vary based on whether the API is being accessed via kong or service directly.

Reply:
Review comment incorporated in the script. Variables has been introduced for all api endpoints